From: Ben Hutchings <ben@decadent.org.uk>
Subject: Clean modules without link-vmlinux.sh
Date: Sat, 29 Dec 2012 01:51:05 +0100
Bug-Debian: http://bugs.debian.org/696775

Calling link-vmlinux.sh when cleaning external modules is stupid.

--- a/Makefile
+++ b/Makefile
@@ -1036,11 +1036,13 @@ clean: rm-dirs  := $(CLEAN_DIRS)
 clean: rm-files := $(CLEAN_FILES)
 clean-dirs      := $(addprefix _clean_, . $(vmlinux-alldirs) Documentation samples)
 
-PHONY += $(clean-dirs) clean archclean
+PHONY += $(clean-dirs) clean-vmlinux clean archclean
 $(clean-dirs):
 	$(Q)$(MAKE) $(clean)=$(patsubst _clean_%,%,$@)
+clean-vmlinux:
+	$(Q)$(CONFIG_SHELL) $(srctree)/scripts/link-vmlinux.sh clean
 
-clean: archclean
+clean: clean-vmlinux archclean
 
 # mrproper - Delete all generated files, including .config
 #
@@ -1267,7 +1269,6 @@ scripts: ;
 endif # KBUILD_EXTMOD
 
 clean: $(clean-dirs)
-	$(Q)$(CONFIG_SHELL) $(srctree)/scripts/link-vmlinux.sh clean
 	$(call cmd,rmdirs)
 	$(call cmd,rmfiles)
 	@find $(if $(KBUILD_EXTMOD), $(KBUILD_EXTMOD), .) $(RCS_FIND_IGNORE) \
